### PR TITLE
Add specifications for ExperimentalDetectron* operations

### PR DESCRIPTION
**Pull request by achetver**
### Details:
 - Added specification for next operations:
   - ExperimentalDetectronDetectionOutput-6;
   - ExperimentalDetectronGenerateProposalsSingleImage-6;
   - ExperimentalDetectronDetectionOutput-6;
   - ExperimentalDetectronROIFeatureExtractor-6;
   - ExperimentalDetectronTopKROIs-6.
 

### Tickets:
 - 43924

### CODE REVIEW

File: docs/ops/experimental/ExperimentalDetectronDetectionOutput_6.md
Comment (vgavrilo) [inline docs/ops/experimental/ExperimentalDetectronDetectionOutput_6.md]: Please, write examples.

File: docs/ops/experimental/ExperimentalDetectronDetectionOutput_6.md
Comment (vgavrilo) [inline docs/ops/experimental/ExperimentalDetectronDetectionOutput_6.md]: Please, add detailed descriptions of these inputs.

**Review by vgavrilo — CHANGES_REQUESTED** [review]

File: docs/ops/experimental/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/experimental/ExperimentalDetectronDetectionOutput_6.md]: It was template, added by a mistake. Now it is removed.

File: docs/ops/experimental/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/experimental/ExperimentalDetectronDetectionOutput_6.md]: It was template, added by a mistake. Now it is removed.

File: docs/ops/experimental/ExperimentalDetectronROIFeatureExtractor_6.md
Comment (vgavrilo) [inline docs/ops/experimental/ExperimentalDetectronROIFeatureExtractor_6.md:1]: Maybe you should write the PR title as 'Add specification for `ExperimentalDetectronROIFeatureExtractor`, because this PR doesn't contain other Detectron-like operations specifications?

Comment (vgavrilo) [reply docs/ops/experimental/ExperimentalDetectronROIFeatureExtractor_6.md:1]: If previous my comment is not correct, please, add other specifications.

**Review by vgavrilo — CHANGES_REQUESTED** [review]

File: docs/ops/experimental/ExperimentalDetectronROIFeatureExtractor_6.md
Comment (achetver) [reply docs/ops/experimental/ExperimentalDetectronROIFeatureExtractor_6.md:1]: All operations specifications will be added in this review. I will add rest operations without detailed descriptions today or tomorrow.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:26]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:33]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:42]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:50]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:58]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:75]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:67]: Maybe it is not required attribute?

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:84]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md:27]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md:35]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md:42]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md:51]: If this attribute is required, the we needn't specify the default value.

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md:19]: It is optional attribute, see https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/vpu/graph_transformer/src/stages/exp_priorgridgenerator.cpp#L77

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md:27]: Could you write descriptions of `h` and `w` as separate descriptions? And more, these attributes are optional, see lines 78 and 79 of the file https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/vpu/graph_transformer/src/stages/exp_priorgridgenerator.cpp.

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md:35]: Could you write descriptions of `stride_x` and `stride_y` as separate descriptions? And more, these attributes are optional, see lines 80 and 81 of the file https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/vpu/graph_transformer/src/stages/exp_priorgridgenerator.cpp.

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md:64]: Maybe these lines should be formatted as C++ code?

File: docs/ops/detection/ExperimentalDetectronTopKROIs_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronTopKROIs_6.md:18]: It is an optional attribute (see https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/vpu/graph_transformer/src/stages/exp_topkrois.cpp#L65).

**Review by vgavrilo — CHANGES_REQUESTED** [review]

**Comment by lazarevevgeny** [comment]
@achetver , once all comments are resolved we should open a PR to releases/2021/3 branch

**Comment by achetver** [comment]
@lazarevevgeny Ok.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:26]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:42]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:50]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:58]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:67]: Yes, updated.

File: docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md:27]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md:35]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md:51]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md]: Yes, updated.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:84]: Sure, deleted.

File: docs/ops/detection/ExperimentalDetectronTopKROIs_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronTopKROIs_6.md:18]: Ok, updated.

File: ngraph/core/include/ngraph/op/experimental_detectron_detection_output.hpp
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/experimental_detectron_detection_output.hpp:31]: Why did you remove doxygen documentation?

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md]: Done.

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md]: Updated, but this part is still uses pseudocode-like style. Do you think C++ formatting will be better to understanding?

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md]: Done.

File: ngraph/core/include/ngraph/op/experimental_detectron_detection_output.hpp
Comment (achetver) [reply ngraph/core/include/ngraph/op/experimental_detectron_detection_output.hpp:31]: Because links to openvino/training_extensions repository is outdated. Files mentioned in these descriptions were removed in https://github.com/openvinotoolkit/training_extensions/commit/e7eed680b35acd0bac2060ac1649bea1d76d33c0

**Comment by achetver** [comment]
@vgavrilo @rkazants @lazarevevgeny please, review these specifications.

**Comment by achetver** [comment]
@openvinotoolkit/openvino-docs-maintainers Please, start review these changes.

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (vgavrilo) [reply docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md]: If we write this C++-code, then we should use Markdown tags to syntax highlight, that is
```c++
    for (int ih = 0; ih < layer_height; ++ih)
        for (int iw = 0; iw < layer_width; ++iw)
            for (int s = 0; s < number_of_priors; ++s)
                output_data[0] = priors[4 * s + 0] + step_w * (iw + 0.5)
                output_data[1] = priors[4 * s + 1] + step_h * (ih + 0.5)
                output_data[2] = priors[4 * s + 2] + step_w * (iw + 0.5)
                output_data[3] = priors[4 * s + 3] + step_h * (ih + 0.5)
                output_data += 4
```
But we can replace this code with some simple description, as @lazarevevgeny requested

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md:17]: If we write this C++-code, the we should to use Markdown tags to syntax highlight, that is
```c++
    for (int ih = 0; ih < layer_height; ++ih)
        for (int iw = 0; iw < layer_width; ++iw)
            for (int s = 0; s < number_of_priors; ++s)
                output_data[0] = priors[4 * s + 0] + step_w * (iw + 0.5)
                output_data[1] = priors[4 * s + 1] + step_h * (ih + 0.5)
                output_data[2] = priors[4 * s + 2] + step_w * (iw + 0.5)
                output_data[3] = priors[4 * s + 3] + step_h * (ih + 0.5)
                output_data += 4
```
But, according to @lazarevevgeny, we can write simple description, without C++-code.

**Review by vgavrilo — CHANGES_REQUESTED** [review]

File: ngraph/core/include/ngraph/op/experimental_detectron_detection_output.hpp
Comment (ilyachur) [reply ngraph/core/include/ngraph/op/experimental_detectron_detection_output.hpp:31]: I propose to add the new description for the class but don't remove full information about the operation.
If we have doxygen description for the operation, it allows to have generate API documentation for the op.

File: docs/ops/detection/ExperimentalDetectronROIFeatureExtractor_6.md
Comment (lazarevevgeny) [inline docs/ops/detection/ExperimentalDetectronROIFeatureExtractor_6.md:17]: Lets wrap these lines with ` characters as we do for most formulas

File: docs/ops/detection/ExperimentalDetectronROIFeatureExtractor_6.md
Comment (lazarevevgeny) [inline docs/ops/detection/ExperimentalDetectronROIFeatureExtractor_6.md:21]: the same comment as above

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (lazarevevgeny) [reply docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md:17]: Let's write simple text description instead of this code.

**Review by lazarevevgeny — APPROVED** [review]
I don't have major technical comments so approve this. But there are a lot of stylistic and grammatical issues which I expect to be fixed by @andrew-zaytsev , @ntyukaev , @aalborov . Please, help us to do this before merge to 2021.3 release branch!

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (rkazants) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:93]: `number_of_POIs` -> `number_of_ROIs`

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (rkazants) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:75]: It seems to combine two attributes. Let us decouple them

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (rkazants) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:20]: Not clear how `post_nms_count` participates here. Please describe it.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (rkazants) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:50]: `sampling_ratio` or `post_nms_count`?

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (rkazants) [inline docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:12]: Do you mean `delta_weights`? Could you please elaborate how they are applied?

File: docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md
Comment (rkazants) [inline docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md:20]: the rest of the output is filled by zeros

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:75]: No, this is one attribute according to source code: https://github.com/openvinotoolkit/openvino/blob/83e5bde4ea9ee7d91b852ecc9707b78b01fcc7f1/ngraph/core/include/ngraph/op/experimental_detectron_detection_output.hpp#L47

And we decide to not update exist operations, just add specifications to them as is.

File: docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronGenerateProposalsSingleImage_6.md:20]: No, in this case operation returns two tensors filled by zeroes, see code below in operation realization: https://github.com/openvinotoolkit/openvino/blob/83e5bde4ea9ee7d91b852ecc9707b78b01fcc7f1/inference-engine/src/mkldnn_plugin/nodes/proposal_onnx.cpp#L264

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md]: Done.

File: ngraph/core/include/ngraph/op/experimental_detectron_detection_output.hpp
Comment (achetver) [reply ngraph/core/include/ngraph/op/experimental_detectron_detection_output.hpp:31]: Ok, I return and update these descriptions.

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md]: Ok, done.

File: docs/ops/detection/ExperimentalDetectronROIFeatureExtractor_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronROIFeatureExtractor_6.md]: Done.

File: docs/ops/detection/ExperimentalDetectronROIFeatureExtractor_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronROIFeatureExtractor_6.md]: Done.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md]: Done.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:11]: Ok, description updated.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md:54]: Updated.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md]: Done.

File: docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronDetectionOutput_6.md]: `post_nms_count` of course, fixed.

**Comment by achetver** [comment]
@andrew-zaytsev , @ntyukaev , @aalborov , please, take a look and review these changes in documentation.

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (vgavrilo) [inline docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md:13]: Maybe 'cells', not 'sells'?

**Review by vgavrilo — CHANGES_REQUESTED** [review]

File: docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md
Comment (achetver) [reply docs/ops/detection/ExperimentalDetectronPriorGridGenerator_6.md]: Yes, this typo fixed.

**Review by vgavrilo — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**
